Paco-Server/src/com/google/sampling/experiential/server/EventRetriever.java

```diff
@@ -83,31 +90,43 @@ public static synchronized EventRetriever getInstance() {
     }
     return instance;
   }
-
-  public void postEvent(String who, String lat, String lon, Date whenDate, String appId, String pacoVersion,
+  
+  public void postEvent(boolean persistInCloudSql, JSONObject eventJson, String who, String lat, String lon, Date whenDate, String appId, String pacoVersion,
                         Set<What> whats, boolean shared, String experimentId, String experimentName,
                         Integer experimentVersion, DateTime responseTime, DateTime scheduledTime,
                         List<PhotoBlob> blobs, String groupName, Long actionTriggerId, Long actionTriggerSpecId, Long actionId) {
-
     final String tz = responseTime != null && responseTime.getZone() != null
             ? responseTime.getZone().toString()
             : scheduledTime!= null && scheduledTime.getZone() != null
               ? scheduledTime.getZone().toString()
               : null;
-    postEvent(who, lat, lon, whenDate, appId, pacoVersion, whats, shared, experimentId, experimentName, experimentVersion,
+              
+     postEvent(persistInCloudSql, eventJson, who, lat, lon, whenDate, appId, pacoVersion, whats, shared, experimentId, experimentName, experimentVersion,
               responseTime != null ? responseTime.toDate() : null,
               scheduledTime != null ? scheduledTime.toDate() : null, blobs,
               tz, groupName, actionTriggerId, actionTriggerSpecId, actionId);
+  
+  }
+    
+
+  public void postEvent(String who, String lat, String lon, Date whenDate, String appId, String pacoVersion,
+                        Set<What> whats, boolean shared, String experimentId, String experimentName,
+                        Integer experimentVersion, DateTime responseTime, DateTime scheduledTime,
+                        List<PhotoBlob> blobs, String groupName, Long actionTriggerId, Long actionTriggerSpecId, Long actionId) {
+    
+    postEvent(true, null, who, lat, lon, whenDate, appId, pacoVersion, whats, shared, experimentId, experimentName, 
+                     experimentVersion, responseTime, scheduledTime, blobs, groupName, actionTriggerId, actionTriggerSpecId, actionId);
+   
   }
 
 
-  public void postEvent(String who, String lat, String lon, Date whenDate, String appId,
+  public void postEvent(boolean persistInCloudSql, JSONObject eventJson, String who, String lat, String lon, Date whenDate, String appId,
```

Comment:
It looks like the persistInCloudSql means persistInCloudSqlOnly. Is that true? If so, please rename it to clarify that.